web/src/emoji_picker.js

```diff
@@ -286,7 +287,7 @@ function toggle_reaction(emoji_name, event) {
         return;
     }
 
-    const message_id = current_message_id;
+    const message_id = message_lists.current.selected_id();
     const message = message_store.get(message_id);
```

Comment:
I don't think this is a correct change since we are tracking for which message id the popover is open for which can be different from the selected message.

Reply:
Since these changes were made quite some time ago, I can't recall why exactly I did them. For now, I've reverted them back as they appeared to have no impact on the functionality of my pull request.